includes/class-nrh.php

```diff
@@ -17,99 +17,451 @@ class NRH {
-		$hosts[] = 'checkout.fundjournalism.org';
-		return $hosts;
+	public static function get_clean_site_url() {
+		$protocols = [ 'http://', 'https://' ];
+		return str_replace( $protocols, '', get_bloginfo( 'wpurl' ) );
 	}
 
 	/**
-	 * Add a hidden campaign input when a custom campaign is present in the GET request.
+	 * Rewrite Donate block for News Revenue Hub submission.
 	 *
 	 * @param string $html The donate form html.
+	 * @param array  $attributes Block attributes.
+	 *
 	 * @return string modified $html.
 	 */
-	public static function handle_custom_campaign_id( $html ) {
-		// Don't add a global campaign ID if there is already a campaign ID.
-		if ( stripos( $html, "name='campaign'" ) || stripos( $html, 'name="campaign"' ) ) {
-			return $html;
-		}
+	public static function render_nrh_donate_block( $html, $attributes ) {
+			$settings = array_merge( Donations::get_donation_settings(), $attributes );
 
-		$custom_campaign = filter_input( INPUT_GET, 'campaign', FILTER_SANITIZE_STRING );
-		if ( ! $custom_campaign ) {
-			return $html;
-		}
+			$frequencies = [
+				'once'  => __( 'One-time', 'newspack-blocks' ),
+				'month' => __( 'Monthly', 'newspack-blocks' ),
+				'year'  => __( 'Annually', 'newspack-blocks' ),
+			];
 
-		$custom_campaign = '<input type="hidden" name="campaign" value="' . esc_attr( $custom_campaign ) . '"/>';
-		$html            = str_replace( '</form>', $custom_campaign . '</form>', $html );
-		return $html;
+			$nrh_frequency_keys = [
+				'once'  => 'once',
+				'month' => 'monthly',
+				'year'  => 'yearly',
+
+			];
+
+			$selected_frequency = isset( $attributes['defaultFrequency'] ) ? $attributes['defaultFrequency'] : 'month';
+			$suggested_amounts  = $settings['suggestedAmounts'];
+
+			$campaign = isset( $attributes['campaign'] ) ? $attributes['campaign'] : false;
+
+			$uid = wp_rand( 10000, 99999 ); // Unique identifier to prevent labels colliding with other instances of Donate block.
+
+			$button_text = $attributes['buttonText'];
+
+			$nrh_config = get_option( NEWSPACK_NRH_CONFIG );
+
+			$organization_id = wp_strip_all_tags( $nrh_config['nrh_organization_id'] );
+
+			$allowed_tags = [
+				'form'  => [
+					'method' => [],
+					'action' => [],
+				],
+				'input' => [
+					'type'  => [],
+					'name'  => [],
+					'value' => [],
+				],
+			];
+
+			$is_amp        = function_exists( 'is_amp_endpoint' ) && is_amp_endpoint();
+			$block_id      = 'wpbdbd_' . uniqid();
+			$initial_value = $suggested_amounts[1];
+			if ( ! $settings['tiered'] ) {
+				$initial_value = $settings['suggestedAmountUntiered'];
+				if ( in_array( $selected_frequency, [ 'year', 'once' ] ) ) {
+					$initial_value *= 12;
+				}
+			}
+
+			$form_open = sprintf(
+				'<form method="GET" action="https://checkout.fundjournalism.org/memberform"><input type="hidden" name="org_id" value="%s" />',
+				esc_attr( $organization_id )
+			);
+
+			// phpcs:disable
+
+			ob_start();
+
+		if ( ! $is_amp ) : ?>
```

Comment:
I only noticed when I reached this line, but a lot of this function is indented weirdly. Above this, everything is indented too much. Below this, the matching `endif` doesn't line up with the opening `if ( ! $is_amp ) :`.

Reply:
Here's the saga. All of the weird indentation was done to comply with PHPCS rules, which get pretty odd when dealing with large blocks of markup. At some point I ended up adding PHPCS ignore rules, but left the indentation intact. In https://github.com/Automattic/newspack-plugin/pull/810/commits/1fa6e2950baaa5c2277d8ed8d295f6cd1a717c5b I was just able to resolve a couple remaining PHPCS issues and restored PHPCS. If you check it now you'll see that any change to the indentation will be flagged by PHPCS.